core/include/drivers/regulator.h

```diff
@@ -71,14 +71,25 @@ enum voltage_type {
  * @type: Type of level description
  * @num_levels: Number of cells of @entries when @type is VOLTAGE_TYPE_FULL_LIST
  * @entries: Voltage level information in uV
- *
  */
 struct regulator_voltages {
 	enum voltage_type type;
 	size_t num_levels;
 	int entries[];
```

Comment:
Why aren't we using `int entries[3];` to skip all the workarounds?

Reply:
because there may be more than 3 entries, or less. At least one.

Reply:
How come that `struct voltages_fallback` works then?

Reply:
`struct voltages_fallback` has 3 preallocated level cells because it's used with either 3 cells (min/max/step case) or with 1 cell (fixed regulator).

For info, this voltage level description is related to how the SCMI spec expects those data to be formated. Actually the SCMI server is the only user of the `regulator_supported_voltages()` API function and related `struct regulator_ops::supported_voltages` operation.

Reply:
Having a variable-sized struct embedded in the middle of another struct is begging for problems and non-obvious workarounds. I suggest either using a pointer to an appropriately allocated instance of `struct regulator_voltages`, or finding a reasonable maximum value and turning it into a fixed-sized struct.

Reply:
With the latest fixup commits I've added, the variable sized `struct regulator_voltages` no more embedded inside another structure, it's only used as argument in API function `regulator_supported_voltages()` and in operation handler functions `struct regulator_ops::supported_voltages`. Do you think it's ok like that?

Reply:
I still don't understand why it's OK to hardcode the range at one place, but not at another. You seem to want the flexibility of variable size, but you implement it using fixed-size workarounds. Why? To save memory or to keep things "simple"?

Reply:
I'll simplify things. Indeed the implementation i made was not very nice. In progress.